ts/WoltLabSuite/Core/Ui/Poll/Manager/Poll.ts

```diff
@@ -0,0 +1,205 @@
+/**
+ * Handles the poll UI.
+ *
+ * @author  Joshua Ruesweg
+ * @copyright  2001-2022 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Poll/Manager/Poll
+ * @since   5.5
+ */
+
+import DomChangeListener from "../../../Dom/Change/Listener";
+import DomUtil from "../../../Dom/Util";
+import { formatNumeric } from "../../../StringUtil";
+import Participants from "./View/Participants";
+import Results from "./View/Results";
+import VoteView from "./View/Vote";
+import VoteHandler from "./Vote";
+
+export enum PollViews {
+  vote = "vote",
+  results = "results",
+}
+
+export class Poll {
+  public readonly pollID: number;
+  protected poll: HTMLElement;
+
+  private voteView?: VoteView;
+  private resultsView?: Results;
+  private participants?: Participants;
+
+  private voteHandler?: VoteHandler;
```

Comment:
```suggestion
  private readonly voteView?: VoteView = undefined;
  private readonly resultsView?: Results = undefined;
  private readonly participants?: Participants = undefined;

  private readonly voteHandler?: VoteHandler = undefined;
```
First of all, all these members are only written inside the constructor, it is safe to mark them as `readonly`.

Second, and I really can't stress this enough, you must always initialize members, otherwise these do not exist on runtime and you will face all sort of strange prototype behaviors. Since their initialization is conditional, you must explicitly define them as `undefined` in their definition!